docs/src/modules/java/pages/quickstart/cr-value-entity-kafka-java.adoc

```diff
@@ -0,0 +1,154 @@
+. Add declarations for:
++
+* Listening to `customer.domain.CustomerState` events from the value entity `customer` by using the option `eventing.in`.
+* Publishing to the Kafka topic `customer_upserts` by using the option `eventing.out`.
++
+[source,proto,indent=0]
+.src/main/proto/customer/action/customer_action.proto
+----
+include::example$java-customer-registry-kafka-quickstart/src/main/proto/customer/action/customer_action.proto[tag=service]
+----
+
+. Run `mvn compile` from the project root directory to generate source classes in which you add business logic.
++
+----
+mvn compile
+----
+
+== Create an Action
+
+Actions are stateless functions that can be triggered in multiple ways. In this case the action is triggered by each event `customer.domain.CustomerState` received by the Value Entity `customer.domain.Customer`. 
+
+. If it's not open already, open `src/main/java/customer/action/CustomerStateSubscriptionAction.java` for editing.
+
+. Modify the `onUpsertState` method by adding the logic to handle the action. The complete method should include the following:
++
+[source, java]
+.src/main/java/customer/action/CustomerStateSubscriptionAction.java
+----
+include::example$java-customer-registry-kafka-quickstart/src/main/java/customer/action/CustomerStateSubscriptionAction.java[tag=upsert]
+----
++
+* The incoming message contains the request data from your client and this action just passes it along to the xref:java:actions-publishing-subscribing.adoc[Pub/Sub] mechanism of choice. In this example Kafka.
+
+== Package and deploy your service
+
+To build and publish the container image and then deploy the service, follow these steps:
+
+. If you haven't done so yet, sign in to your Akka Serverless account. If this is your first time using Akka Serverless, this will let you register an account, https://developer.lightbend.com/docs/akka-serverless/projects/create-project.html[create your first project], and set this project as the default.
++
+----
+akkasls auth login
+----
+
+. Use the `deploy` target to build the container image, publish it to the container registry as configured in the `pom.xml` file, and then automatically https://developer.lightbend.com/docs/akka-serverless/services/deploy-service.html#_deploy[deploy the service] to Akka Serverless using `akkasls`:
++
+----
+mvn deploy
+----
+
+. You can https://developer.lightbend.com/docs/akka-serverless/services/deploy-service.html#_verify_service_status[verify the status of the deployed service] using:
++
+----
+akkasls service list
+----
+
+== Invoke your service
+
+Now that you have deployed your service, the next step is to invoke it using gRPCurl
+
+. From the "_Service Explorer_" click on the method you want to invoke
+. Click on "_gRPCurl_"
+. In the bottom section of the dialog, fill in the values you want to send to your service
+. In the top section of the dialog, click the "_Copy to clipboard_" button
+. Open a new command line and paste the content you just copied
+
```

Comment:
It would probably be nice to be able to see that it worked. Add some hint about looking at the topic to see that the message arrived. Confluent Cloud has a nice console to inspect the topics.

Reply:
done